File: app/src/main/java/com/ufkoku/ciphersharedpreferences/demo/ui/MainViewState.java
package com.ufkoku.ciphersharedpreferences.demo.ui;

import com.ufkoku.ciphersharedpreferences.demo.entity.PrefsEntry;
import com.ufkoku.mvp_base.viewstate.IViewState;

import java.util.ArrayList;
import java.util.List;

public class MainViewState implements IViewState<IMainActivity> {

    private List<PrefsEntry> entries = new ArrayList<>();

    private boolean prefsLoaded = false;

    public List<PrefsEntry> getEntries() {
        return entries;
    }

    public boolean isPrefsLoaded() {
        return prefsLoaded;
    }

    public void setPrefsLoaded(boolean prefsLoaded) {
        this.prefsLoaded = prefsLoaded;
    }

    @Override
    public void apply(IMainActivity activity) {
        if (prefsLoaded) {
            activity.populateEntries(entries);
        }
    }

}

File: app/src/main/kotlin/com/ufkoku/ciphersharedpreferences/demo/ui/main/MainViewState.kt
package com.ufkoku.ciphersharedpreferences.demo.ui.main

import android.os.Bundle

import com.ufkoku.ciphersharedpreferences.demo.entity.PrefsEntry
import com.ufkoku.mvp.viewstate.autosavable.AutoSavable
import com.ufkoku.mvp.viewstate.autosavable.DontSave
import com.ufkoku.mvp_base.viewstate.IViewState

import java.util.ArrayList

@AutoSavable
class MainViewState : IViewState<MainViewState.IViewStateHolder> {

    var key: String? = null

    @DontSave
    var arePrefsLoaded = false

    @DontSave
    val entries: MutableList<PrefsEntry> = ArrayList()

    override fun restore(inState: Bundle) {
        MainViewStateSaver.restore(this, inState)
    }

    override fun save(out: Bundle) {
        MainViewStateSaver.save(this, out)
    }

    override fun apply(view: IViewStateHolder) {
        if (arePrefsLoaded) {
            view.populateEntries(entries)
        }
    }

    interface IViewStateHolder {

        fun populateEntries(entries: List<PrefsEntry>)

    }

}
